### PR TITLE
Add proper fields ordering during schemas JSON marshalling

### PR DESCRIPTION
**Pull request by morzhanov**
## Issue
Golang scrambles the JSON output, it needs to be ordered correctly before returning.

## Solution
Ordered JSON serialized content output: 
- [x] updated all Marshal methods with ordered maps to preserve proper fields ordering in schemas
- [x] updated serializer to handle new methods

### CODE REVIEW

File: workflowtrigger/util/auditlog/log.go [hunk 1]
```diff
@@ -113,23 +115,38 @@ func UnmarshallString(str string) Log {
...
 	case *RemediationLog:
-		mapObj := serializerutil.GetMapFromStruct(log)
-		mapObj["type"] = serializerutil.RemediationLogType
-		return mapObj
+		remLog := log.(*RemediationLog)
+		mapObj.Set("pipelineUniqueVersionNumber", remLog.PipelineUniqueVersionNumber)
+		mapObj.Set("uniqueVersionInstance", remLog.UniqueVersionInstance)
+		mapObj.Set("unhealthyResources", remLog.UnhealthyResources)
+		mapObj.Set("remediationStatus", remLog.RemediationStatus)
+		mapObj.Set("type", serializerutil.RemediationLogType)
```
Comment (mihirpandya-greenops) [inline workflowtrigger/util/auditlog/log.go:127]: For the remediation log and the deployment log, put the type as the first component.

File: workflowtrigger/util/clientrequest/request_events.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package clientrequest
```
Comment (mihirpandya-greenops) [inline workflowtrigger/util/clientrequest/request_events.go:1]: For all the serialization in these methods, order does not matter. Is it possible to automatically insert all the keys using the OrderedMap (not add them explicitly)?

File: workflowtrigger/util/event/kafka_events.go [hunk 3]
```diff
@@ -2,6 +2,7 @@ package event
```
Comment (mihirpandya-greenops) [inline workflowtrigger/util/event/kafka_events.go:2]: For all the serialization in these methods, order does not matter. Is it possible to automatically insert all the keys using the OrderedMap (not add them explicitly)?

**Review by mihirpandya-greenops — APPROVED** [review]
Remove all the .DS_Store files from the PR

File: workflowtrigger/util/auditlog/log.go (hunk 1 above)
Comment (morzhanov) [reply workflowtrigger/util/auditlog/log.go]: Moved `type` field on the first place in those objects.

File: workflowtrigger/util/clientrequest/request_events.go (hunk 2 above)
Comment (morzhanov) [reply workflowtrigger/util/clientrequest/request_events.go]: Unfortunately, OrderedMap does not have a method for bulk insertion. If we do not need those objects to be ordered we can use a simple map here. Updated those methods with `map[string]interface{}` return value.

File: workflowtrigger/util/event/kafka_events.go (hunk 3 above)
Comment (morzhanov) [reply workflowtrigger/util/event/kafka_events.go]: Updated those methods with `map[string]interface{}` return value.

**Comment by morzhanov** [comment]
> Remove all the .DS_Store files from the PR

Yeah, those are annoying mac os auto-generated files. Removed those files and added `.DS_Store` to the `gitignore`.

**Review by mihirpandya-greenops — APPROVED** [review]
Looks great!

**Pull request merged**